test

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh -eu
+
+add_test() {
+    _unit_tests="${_unit_tests}:$1"
+}
+
+dirname() {
+    _dir="${1%/}"
```

Comment:
Do we want to strip all trailing `/` or just the last one?

Compare:

```
_dir=fo///
echo ${_dir%/}
echo ${_dir%%"${_dir##*[!/]}"}
```

Not certain if it matters but I also don't know why implement `dirname` in `sh` if we already have `/usr/bin/dirname`.